core/server/api/themes.js

```diff
@@ -21,41 +24,57 @@ themes = {
      * @returns {Promise(Themes)}
      */
     browse: function browse(options) {
-        options = options || {};
+        var tasks;
 
-        return canThis(options.context).browse.theme().then(function () {
+        function handlePermissions(options) {
+            return canThis(options.context).browse.theme().then(function () {
+                return Promise.resolve();
+            }, function () {
+                return Promise.reject(new errors.NoPermissionError('You do not have permission to browse themes.'));
```

Comment:
This one should be in the form `return errors.handleAPIError(error, 'message here');` I think